src/Operation/Watch.php

```diff
@@ -376,7 +378,7 @@ private function resume($resumeToken = null, $hasAdvanced = false)
         $this->hasResumed = true;
 
         // Select a new server using the original read preference
-        $server = $this->manager->selectServer($this->aggregateOptions['readPreference']);
+        $server = select_server($this->manager, $this->aggregateOptions['readPreference'], extract_session_from_options($this->aggregateOptions));
```

Comment:
A watch operation is currently not allowed in a transaction:
> MongoDB\Driver\Exception\CommandException : Operation not permitted in transaction :: caused by :: Operation not permitted in transaction :: caused by :: $changeStream cannot run with a readConcern other than 'majority', or in a multi-document transaction. Current readConcern: { readConcern: { level: "snapshot" } }

I originally changed all instances of server selection within the library with the new logic so that it's consistent. In this case, changing server selection back would mean that running the watch operation on a pinned session will error because we're running the command on the wrong server, instead of informing the user that the operation can't be done in a transaction. Even in this technically impossible case, I'd still pick the pinned server to have the correct exception shown to the user.

This will also avoid problems down the line if watch ever becomes available in a transaction.

Reply:
I concur that this would ensure the user gets more accurate error message. Would you mind adding a comment to document this reasoning?